src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/VerifyUserWithTypingDNAFunctionImpl.java

```diff
@@ -0,0 +1,163 @@
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.CommonUtils;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.net.ssl.HttpsURLConnection;
+
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants.OUTCOME_FAIL;
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants.OUTCOME_SUCCESS;
+
+/**
+ * Custom adaptive function implementation for verifying
+ * users typing pattern with TypingDNA.
+ */
+
+public class VerifyUserWithTypingDNAFunctionImpl implements VerifyUserWithTypingDNAFunction {
+
+    private static final Log log = LogFactory.getLog(VerifyUserWithTypingDNAFunctionImpl.class);
+
+    @Override
+    public void verifyUserWithTypingDNA(JsAuthenticationContext context, Map<String, Object> eventHandlers) throws Exception {
+
+        JsAuthenticatedUser user = utils.getUser(context);
+        String username = user.getWrapped().getUserName();
+        String tenantDomain = user.getWrapped().getTenantDomain();
+        String typingPattern = utils.getTypingPattern(context);
+
+        AtomicReference<Long> eCode = new AtomicReference<>(0L);
+        AtomicBoolean result = new AtomicBoolean(false);
+
+        //getting connector configurations
+        String APIKey = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.USERNAME, tenantDomain);
+        String APISecret = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.CREDENTIAL, tenantDomain);
+        String advanced = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.ADVANCE_MODE_ENABLED, tenantDomain);
+        String region = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.REGION, tenantDomain);
+
+        String userID = DigestUtils.sha256Hex(username + "@" + tenantDomain);
+
+        //selecting the suitable typingDNA API according to the configuration
+        String api = advanced.equals(Constants.TRUE) ? Constants.VERIFY_API : Constants.AUTO_API;
+
+        AsyncProcess asyncProcess = new AsyncProcess((authenticationContext, asyncReturn) -> {
+            try {
+                if (!(typingPattern == null) && !typingPattern.equals(Constants.NULL)) {
+                    //set typing patterns received true
```

Comment:
Comments should start with space and an upper case letter and end with a full stop. Check all the places and fix formatting.